### ISSUE TITLE
False positive and false fix for DisallowYodaComparison

### ISSUE THREAD

**Issue by herndlm** [slevomat/coding-standard#1658]
The following PHP HTML template code causes a false positive via `DisallowYodaComparison`:

```php
<input
    type="radio"
    <?= $disabledInputs ? 'disabled' : '' ?>
    <?= $policy === Policy::PUBLIC->value ? 'checked' : '' ?>
    value="<?= esc_attr(Policy::PUBLIC->value) ?>">
```

![image](https://github.com/slevomat/coding-standard/assets/5738896/909814c1-0c64-4525-b4f6-febdb1038018)

even more interesting though maybe, if I try to auto-fix this I end up with the following which is syntactically very different:
```php
<input
    type="radio"
    <?= $disabledInputs ? 'disabled' : Policy::PUBLIC->value === '' ?>
    <?= $policy ? 'checked' : '' ?>
    value="<?= esc_attr(Policy::PUBLIC->value) ?>">
```

it looks like the 2 independent ternaries are somehow influencing each other. adding a semicolon after the first ternary seems to work around it.

**Comment by kukulich** [comment]
Our sniffs are not tested for PHP templates. I'm ok to merge PR that will fix it but we will not try to fix it ourself.
